Hw5/Task1/SparseMatrix.cs

```diff
@@ -0,0 +1,128 @@
+using System.Collections;
+using System.Text;
+
+public class SparseMatrix : IEnumerable<long>
+{
+    private Dictionary<(int, int), long> nonZero = new();
+    public int Height {get; private set;}
+    public int Width {get; private set;}
+
+    public SparseMatrix(int height, int width)
+    {
+        if(height <= 0 || width <= 0) 
+        {
+            throw new ArgumentException("Neither height nor width can be less or equal to zero");
+        }
+        Height = height;
+        Width = width;
+    }
+    public SparseMatrix(long[,] matrix)
+    {
+        if(matrix.GetLength(0) <= 0 || matrix.GetLength(1) <= 0) 
+        {
+            throw new ArgumentException("Neither height nor width of a matrix can be less or equal to zero");
+        }
+        Height = matrix.GetLength(0);
+        Width = matrix.GetLength(1);
+        for(int i = 0; i < Height; i++)
+        {
+            for(int j = 0; j < Width; j++)
+            {
+                if(matrix[i, j] != 0)
+                {
+                    this[i, j] = matrix[i, j];
+                }
+            }
+        }
+    }
+
+    public long this[int i, int j]
+    {
+        get
+        {
```

Comment:
Move check yo a separate private method not to duplicate code, DRY. Use it both at line  43 and 51